scripts/cicd/verify-cicd-operation.py

```diff
@@ -30,21 +40,29 @@ def runner(program, *args):
     sys.exit(11)
     # runner never returns
 
-logger = logging.getLogger('verify_command_logger')
-logger.setLevel(logging.INFO)
-logger.addHandler(logging.handlers.SysLogHandler('/dev/log'))
-
-keyname = sys.argv[1] if len(sys.argv) >= 2 else "<none specified>"
-cmd = os.environ.get("SSH_ORIGINAL_COMMAND", "")
+def verify(allowed_args, args):
+    """ Check if `args` is valid, according to `allowed_args`. """
+    for allowed in allowed_args:
+        if len(allowed) != len(args):
+            continue
+        if all(re.match(pat + '$', arg) for (pat, arg) in zip(allowed, args)):
+            return True
+    return False
 
-match = re.match(r"(?P<operation>install|delete|upgrade)", cmd)
+if __name__ == '__main__':
+    """ Module entrypoint when executed directly. """
+    logger = logging.getLogger('verify_command_logger')
+    logger.setLevel(logging.INFO)
+    logger.addHandler(logging.handlers.SysLogHandler('/dev/log'))
 
-if match:
-    logger.info("%s Restricted key '%s' running command: %s" % (os.path.basename(__file__), keyname, cmd))
-    runner("/home/opsmedic/aos-cd/git/aos-cd-jobs/bin/cicd-control.sh", keyname, match.group("operation"))
+    keyname = sys.argv[1] if len(sys.argv) >= 2 else "<none specified>"
+    cmd = os.environ.get("SSH_ORIGINAL_COMMAND", "").split()
+    if verify(ALLOWED_COMMANDS, cmd):
+        logger.info("%s Restricted key '%s' running command: %s" % (os.path.basename(__file__), keyname, cmd))
+        runner("/home/opsmedic/aos-cd/git/aos-cd-jobs/bin/cicd-control.sh", keyname, *cmd)
```

Comment:
@joelsmith The RPM work is underway (e.g. #2269 and others ).

keyname is actually a cluster name (e.g. 'dev-preview-int').